python/pyxet/src/lib.rs

```diff
@@ -194,7 +194,11 @@ pub fn perform_mount_curdir(
 #[pymethods]
 impl PyRepoManager {
     #[new]
-    pub fn new() -> PyResult<Self> {
+    pub fn new(domain : &str) -> PyResult<Self> {
+    
+        // A bit of a hack here to make sure the domain gets passed in correctly, 
+        // but hopefully this should be
+        std::env::set_var("XET_ENDPOINT", domain); 
```

Comment:
It creates a completely new XetConfig instance when it's created, so it does seem to pick this up.  I tried a lot of things to get the right endpoint passed down, and doing this finally made it work.  I'd like to change the API to XetRepoManager allow a new endpoint to be passed in on init, but that requires coordinating changes and some work down there.  For now, this seems to work.

Reply:
Hmm, I think urls passed down to xetblob are all parsed format with http(s) schema, so endpoint is already built-in. I suspect this is still used somewhere in pyxet. But it's OK if this makes things work.

Reply:
I thought so, but somewhere it's replacing the endpoint still.  I got everything but some types of downloads to work, and in those it replaced the explicit domain with xethub.com.   So that's a bug, but not sure where.   I think it sets up the repo manager and then that fixes some things?